dist/api_data.py

```diff
@@ -45,6 +45,39 @@ def __ge__(self, other):
 common_runtime_config = [
     Config('app_metadata', '', r'''
         application-owned metadata for this object'''),
+    Config('write_timestamp', 'never', r'''
+        describe how timestamps are expected to be used on modifications
+        to the table. The hint can be used to log verbose messages and/or
+        enforce expected usage. The choices are <code>always</code> to ensure
+        a timestamp is used for every operation on a table,
+        <code>key_consistent</code> to ensure that once timestamps are used for
+        a value they are always used, <code>ordered</code> key_consistent plus
+        subsequent updates to each value must use increasing timestamps,
+        <code>mixed_mode</code> <code>ordered</code> except that updates with
+        no timestamp are allowed and have the effect of resetting the chain of
+        updates once transaction ID based snapshot is no longer relevant.''',
+        choices=['always', 'key_consistent', 'mixed_mode', 'never', 'ordered']),
+    Config('assert', '', r'''
+        enable enhanced checking. ''',
+        type='category', subconfig= [
+        Config('write_timestamp', 'none', r'''
+            verify that commit timestamps are used per the configured
+            <code>write_timestamp</code> option for this table.''',
+            choices=['all', 'diagnostic', 'none']),
+        Config('read_timestamp', 'none', r'''
+            verify that timestamps should 'always' or 'never' be used
+            on reads with this table.  Verification is 'none'
+            if mixed read use is allowed.''',
+            choices=['always', 'never', 'none'])
+        ], undoc=True),
+    Config('verbose', '', r'''
```

Comment:
> I converted the verbose option into a category options type and it works fine.
> That is surprising - I'd attempted to make it consistent with how the verbose option works on the connection handle.

The connection-level options are not saved and passed as an option to the session create, whereas the config options of dhandle or btree are retrieved and used again as options this led to passing `verbose` list option without any options.

> I moved these options into dhandle instead of btree.
> Why did you make this change? Was the information missing from the handle after a restart?

The assert-flags in the btree are used only for storing the assert flags. I added a new flag to store the DHANDLE, so I removed the BTREE assert-flags.

Reply:
I'm going to push back once more on:

> I converted the verbose option into a category options type and it works fine.

I'd prefer it if we could make a list work - that leads to a more consistent user experience.

Reply:
49e956a